tests/test_folder_scanner.py

```diff
@@ -16,30 +16,77 @@ def setUp(self) -> None:
 
     def test_scan_should_detect_entropy_and_not_binary(self):
         folder_path = pathlib.Path(__file__).parent / "data/scan_folder"
+        recurse = True
         self.global_options.entropy = True
         self.global_options.b64_entropy_score = 4.5
         self.global_options.hex_entropy_score = 3
         self.global_options.exclude_signatures = []
         self.global_options.exclude_path_patterns = [r"donotscan\.txt"]
 
-        test_scanner = scanner.FolderScanner(self.global_options, folder_path)
+        test_scanner = scanner.FolderScanner(self.global_options, folder_path, recurse)
         issues = list(test_scanner.scan())
 
-        self.assertEqual(1, len(issues))
-        self.assertEqual("KQ0I97OBuPlGB9yPRxoSxnX52zE=", issues[0].matched_string)
+        self.assertEqual(2, len(issues))
+        actual_issues = []
+        for issue in issues:
+            actual_issues.append(issue.matched_string)
+        self.assertIn("KQ0I97OBuPlGB9yPRxoSxnX52zE=", actual_issues)
+        self.assertIn(
+            "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=",
+            actual_issues,
+        )
         self.assertEqual(IssueType.Entropy, issues[0].issue_type)
 
     def test_scan_should_raise_click_error_on_file_permissions_issues(self):
         folder_path = pathlib.Path(__file__).parent / "data/scan_folder"
+        recurse = True
         self.global_options.entropy = True
         self.global_options.exclude_signatures = []
 
-        test_scanner = scanner.FolderScanner(self.global_options, folder_path)
+        test_scanner = scanner.FolderScanner(self.global_options, folder_path, recurse)
 
         with patch("pathlib.Path.open", side_effect=OSError()):
             with self.assertRaises(click.FileError):
                 list(test_scanner.scan())
 
+    def test_scan_all_the_files_recursively(self):
+        folder_path = pathlib.Path(__file__).parent / "data/scan_folder"
+        recurse = True
+        self.global_options.entropy = True
+        self.global_options.exclude_signatures = []
+        self.global_options.b64_entropy_score = 4.5
+        self.global_options.hex_entropy_score = 3
+
+        test_scanner = scanner.FolderScanner(self.global_options, folder_path, recurse)
+        issues = list(test_scanner.scan())
+
+        self.assertEqual(3, len(issues))
+        actual_issues = []
+        for issue in issues:
+            actual_issues.append(issue.matched_string)
```

Comment:
Same list comprehension trick can be applied here
```suggestion
        actual_issues = [issue.matched_string for issue in issues]
```